cranelift/filetests/filetests/vcode/aarch64/stack.clif

```diff
@@ -0,0 +1,115 @@
+test vcode
+target aarch64
+
+function %stack_addr_small() -> i64 {
+ss0 = explicit_slot 8
+
+block0:
+  v0 = stack_addr.i64 ss0
+  return v0
+}
+
+; check: stp fp, lr, [sp, #-16]!
+; nextln: mov fp, sp
+; nextln: sub sp, sp, #16
+; nextln: load_addr x0, [fp, #-8]
+; nextln: mov sp, fp
+; nextln: ldp fp, lr, [sp], #16
+; nextln: ret
+
+
+function %stack_addr_big() -> i64 {
+ss0 = explicit_slot 100000
+ss1 = explicit_slot 8
+
+block0:
+  v0 = stack_addr.i64 ss0
+  return v0
+}
+
+; check: stp fp, lr, [sp, #-16]!
+; nextln: mov fp, sp
+; nextln: ldr x15, 8 ; b 12 ; data 100016
+; nextln: sub sp, sp, x15, UXTX
+; nextln: movn x15, #34471 ; movk x15, #65534, LSL #16 ; add x15, x15, fp ; load_addr x0, [x15]
```

Comment:
This doesn't match the actually emitted code. The actual code uses `sub` for large negative offsets, while this shows add. This is because I used `mem_finalize_for_show` in the printing code, but manually emitted the right instructions in `emit.rs`. Also the `mem_finalize` code should probably optimize this to a `sub` of the absolute value of the offset when the offset is negative.

Reply:
Fixed